lib/checkunusedvar.cpp

```diff
@@ -1534,7 +1534,43 @@ void CheckUnusedVar::checkStructMemberUsage()
         if (bailout)
             continue;
 
+        // check if class/struct has default equality operator
+        bool has_default_eq = true;
+        for (const Function &f : scope.functionList) {
+            if (f.isOperator() && f.name() == "operator==") {
+                if (f.isDefault()) {
+                    has_default_eq = true;
+                    break;
+                }
+                has_default_eq = false;
+                break;
+            }
+        }
+
+        // check if default equality is used
+        bool default_eq_used = false;
+        if (has_default_eq) {
+            for (const Token *tok = mTokenizer->tokens(); tok; tok = tok->next()) {
+                if (Token::simpleMatch(tok, "==")) {
+                    const Token *lhs = tok->astOperand1();
+                    const Token *rhs = tok->astOperand2();
+                    if (lhs->isVariable() && rhs->isVariable()) {
+                        const Type *ltype = lhs->variable()->type();
+                        const Type *rtype = rhs->variable()->type();
+                        if (ltype && rtype && ltype->name() == scope.className && rtype->name() == scope.className) {
```

Comment:
checking the name will not be robust. "A::X" is the same name as "B::X" because it's not the full scope name. Comparing that the pointers point at the same type/scope is better.